From 23cee39e13cf3f08bdff749b7a1e9720b23df564 Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Tue, 13 Dec 2016 16:57:35 +0100
Subject: [PATCH] joyent/node-manta#294 content-length and transfer-encoding
 chunked must not be used together Reviewed by: David Pacheco <dap@joyent.com>

---
 CHANGES.md    | 4 ++++
 lib/client.js | 9 ++++++++-
 package.json  | 2 +-
 3 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index 8c4b6ee..58b56f4 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,7 +2,11 @@
 
 ## not yet released
 
+## 4.1.1
+
 - joyent/node-manta#293 '~~/' works, '~~' does now too.
+- joyent/node-manta#294 content-length and transfer-encoding chunked must not
+  be used together
 
 ## 4.1.0
 
diff --git a/lib/client.js b/lib/client.js
index 9fa1cea..8de4b24 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -720,8 +720,15 @@ MantaClient.prototype.chattr = function chattr(p, opts, cb) {
      * to be for the PutMetadata request and not metadata to be changed on
      * the Manta object. However a client-side fix is desireable so that
      * 'mchattr' can work with Manta webapis that don't have a fix.
+     *
+     * We only do this if content-length wasn't specified because it's not
+     * legal to specify chunked encoding with content-length. Callers other
+     * than test suites wouldn't normally specify a content length for this
+     * request.
      */
-    options.headers['transfer-encoding'] = 'chunked';
+    if (!options.headers['content-length']) {
+        options.headers['transfer-encoding'] = 'chunked';
+    }
 
     log.debug(options, 'chattr: entered');
     self.signRequest({
diff --git a/package.json b/package.json
index 7d813f2..c36677e 100644
--- a/package.json
+++ b/package.json
@@ -7,7 +7,7 @@
         "type": "git",
         "url": "git://github.com/joyent/node-manta.git"
     },
-    "version": "4.1.0",
+    "version": "4.1.1",
     "main": "./lib/index.js",
     "dependencies": {
         "assert-plus": "^1.0.0",
-- 
2.21.0

